docs/r2/graphs.ipynb

```diff
@@ -0,0 +1,492 @@
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "EGlOqRp54ufD",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "By default, TensorBoard displays the **op-level graph**. (On the left, you can see the “Default” tag selected.)  Note that the graph is inverted; data flows from bottom to top, so it’s upside down compared to the code. However, you can see that the graph closely matches the Keras model definition, with extra edges to other computation nodes.\n",
+        "\n",
+        "Graphs are often very large, so you can manipulate the graph visualization:\n",
+        "\n",
+        "* Scroll to **zoom** in and out\n",
+        "* Drag to **pan**\n",
+        "* Double clicking toggles **node expansion** (a node can be a container for other nodes)\n",
+        "\n",
+        "You can also see metadata by clicking on a node. This allows you to see inputs, outputs, shapes and other details.\n"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "F-2yw5qd7OpK",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "<img class=\"tfo-display-only-on-site\" src=\"https://github.com/tensorflow/tensorboard/blob/master/docs/r2/images/graphs_computation.png?raw=1\"/>"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "jDRynpVw53SJ",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "<img class=\"tfo-display-only-on-site\" src=\"https://github.com/tensorflow/tensorboard/blob/master/docs/r2/images/graphs_computation_detail.png?raw=1\"/>"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "Oj9FSPdz6SO2",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "## Conceptual graph\n",
+        "\n",
+        "In addition to the execution graph, TensorBoard also displays a **conceptual graph**. This is a view of just the Keras model. This may be useful if you’re reusing a saved model and you want to examine or validate its structure.\n",
+        "\n",
+        "To see the conceptual graph, select the “keras” tag. For this example, you’ll see a collapsed **Sequential** node. Double-click the node to see the model’s structure:"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "Qw9rbEcE6eZB",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "<img class=\"tfo-display-only-on-site\" src=\"https://github.com/tensorflow/tensorboard/blob/master/docs/r2/images/graphs_tag_selection.png?raw=1\"/> <br/>\n",
+        "<img class=\"tfo-display-only-on-site\" src=\"https://github.com/tensorflow/tensorboard/blob/master/docs/r2/images/graphs_conceptual.png?raw=1\"/>"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "zVuaKBifu-qF",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "## Graphs of tf.functions\n",
```

Comment:
Does the trace API apply to only tf.function graphs? What if I created a Keras sequential model and trained with `tf.GradientTape()` but not `tf.function` annotation?

Reply:
There's no graph in that situation, so yes, the trace API only applies to tf.function() graphs.